resolwe_bio/processes/workflows/wgs_gvcf.py

```diff
@@ -0,0 +1,376 @@
+"""WGS primary analysis (GVCF)."""
+from resolwe.process import (
+    BooleanField,
+    Data,
+    DataField,
+    FloatField,
+    GroupField,
+    IntegerField,
+    ListField,
+    Process,
+    StringField,
+)
+from resolwe.process.models import Process as BioProcess
+
+
+class WorkflowWgsGvcf(Process):
+    """Whole genome sequencing pipeline (GATK GVCF).
+
+    The pipeline follows GATK best practices recommendations and prepares
+    single-sample paired-end sequencing data for a joint-genotyping step.
+
+    The pipeline steps include Trimmomatic (read trimming), read alignment
+    (BWA-MEM), marking of duplicates (Picard MarkDuplicates), recalibration
+    of base quality scores (ApplyBQSR) and calling of variants
+    (GATK HaplotypeCaller in GVCF mode). The QC reports (FASTQC report,
+    Picard AlignmentSummaryMetrics, CollectWgsMetrics and InsertSizeMetrics)
+    are summarized using MultiQC.
+    """
+
+    slug = "workflow-wgs-gvcf"
+    name = "WGS analysis (GVCF)"
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {
+                "image": "public.ecr.aws/s4q6j6e8/resolwebio/dnaseq:6.0.0",
+            },
+        },
+    }
+    data_name = "WGS GVCF analysis ({{ reads|sample_name|default('?') }})"
+    version = "1.0.0"
+    process_type = "data:workflow:wgs:gvcf"
+    category = "Pipeline"
+    entity = {
+        "type": "sample",
+    }
+
+    class Input:
+        """Input fields."""
+
+        reads = DataField("reads:fastq:paired", label="Input sample")
+        ref_seq = DataField("seq:nucleotide", label="Reference sequence")
+        bwa_index = DataField("index:bwa", label="BWA genome index")
+        known_sites = ListField(
+            DataField("variants:vcf"), label="Known sites of variation (VCF)"
+        )
+
+        advanced = BooleanField(
+            label="Show advanced options",
+            description="Inspect and modify parameters.",
+            default=False,
+        )
+
+        class GatkOptions:
+            """Options."""
+
+            intervals = DataField(
+                "bed",
+                label="Use intervals BED file to limit the analysis to the specified parts of the genome.",
+                required=False,
+            )
+
+            contamination = IntegerField(
+                label="Contamination fraction",
+                default=0,
+                description="Fraction of contamination in sequencing data (for all samples) to aggressively remove.",
```

Comment:
Perhaps break a line to match the width of other lines of code (see for class Trimming).